From 65067bc1ca5c2605c95e79ae9e9ad7fe0504b96c Mon Sep 17 00:00:00 2001
From: Andrei Tatar <andrei@unikraft.io>
Date: Fri, 28 Jul 2023 11:47:32 +0200
Subject: [PATCH] Add compilation guards to SSE2 & SSSE3 code

This change adds compilations guards similar to those in pixman-mmx.c to
prevent compilation of code that uses unsupported processor features.

Signed-off-by: Andrei Tatar <andrei@unikraft.io>
---
 pixman/pixman-sse2.c  | 4 ++++
 pixman/pixman-ssse3.c | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/pixman/pixman-sse2.c b/pixman/pixman-sse2.c
index ce4e75f..28793ba 100644
--- a/pixman/pixman-sse2.c
+++ b/pixman/pixman-sse2.c
@@ -30,6 +30,8 @@
 #include <config.h>
 #endif
 
+#ifdef USE_SSE2
+
 /* PSHUFD is slow on a lot of old processors, and new processors have SSSE3 */
 #define PSHUFD_IS_FAST 0
 
@@ -6526,3 +6528,5 @@ _pixman_implementation_create_sse2 (pixman_implementation_t *fallback)
 
     return imp;
 }
+
+#endif /* USE_SSE2 */
diff --git a/pixman/pixman-ssse3.c b/pixman/pixman-ssse3.c
index 680d6b9..b8613e9 100644
--- a/pixman/pixman-ssse3.c
+++ b/pixman/pixman-ssse3.c
@@ -27,6 +27,8 @@
 #include <config.h>
 #endif
 
+#ifdef USE_SSSE3
+
 #include <stdlib.h>
 #include <mmintrin.h>
 #include <xmmintrin.h>
@@ -349,3 +351,5 @@ _pixman_implementation_create_ssse3 (pixman_implementation_t *fallback)
 
     return imp;
 }
+
+#endif
-- 
2.41.0

